File: p1/069/Solution.java
class Solution {
	public int mySqrt(int x) {
        if (x <= 0) return 0;
        double delta = 0.1d;
        double dx = (double)x;
        double r = dx;
        while(Math.abs(dx - r * r) > delta) {
            r = (x/r + r) / 2.0d;   
        }
        return (int)r;
	}
}

File: p1/069/Solution.scala
object Solution {
    val delta = 0.1d
    def mySqrt(x: Int): Int = {
      if (x <= 0) return 0
      val dx = x.toDouble
      var r = dx
      while ( {
        Math.abs(dx - r * r) > delta
      }) r = (dx / r + r) / 2.0d
      r.toInt
    }
}